NOTES.md

```diff
@@ -0,0 +1,270 @@
+  ```
+
+  _Notes_ => This is the minimal request required => The city field will default to berlin
+  _Backend Response_ =>
+
+  ```
+  {
+    "newBook": {
+      "authors": [
+        "J. K. Rowling"
+      ],
+      "isbn": [],
+      "city": "Berlin",
+      "category": [
+        "fantasy"
+      ],
+      "selectedFiles": [
+        "file 03"
+      ],
+      "reserved": false,
+      "interestedUsers": [],
+      "_id": "60a351257062560d179d7d97",
+      "title": "xmen",
+      "subtitle": "The Big X",
+      "publishedDate": "2002",
+      "pages": 10,
+      "shape": "good condition",
+      "owner": "60a25ada6d711a9e70faa165",
+      "createdAt": "2021-05-18T05:31:17.831Z",
+      "updatedAt": "2021-05-18T05:31:17.831Z"
+    },
+    "addBookToUser": {
+      "avatar": "https://cdn.fakercloud.com/avatars/jjshaw14_128.jpg",
+      "points": 0,
+      "booksToOffer": [
+        "60a25ada6d711a9e70faa16a",
+        "60a25adb6d711a9e70faa16c",
+        "60a25ae16d711a9e70faa177",
+        "60a350be7062560d179d7d96",
+        "60a351257062560d179d7d97"
+      ],
+      "booksToRemember": [
+        "60a25adc6d711a9e70faa16d"
+      ],
+      "booksInterestedIn": [
+        "60a25ae06d711a9e70faa176",
+        "60a25ae16d711a9e70faa178"
+      ],
+      "matches": [],
+      "_id": "60a25ada6d711a9e70faa165",
+      "firstName": "Rickie",
+      "lastName": "Kilback",
+      "username": "Willa.Monahan",
+      "email": "Carlo.Walker@gmail.com",
+      "address": {
+        "country": "Germany",
+        "city": "Berlin"
+      },
+      "createdAt": "2021-05-17T12:00:26.167Z",
+      "updatedAt": "2021-05-18T05:31:18.031Z"
+    }
+  }
+  ```
+
+  _Notes_ => the response: an object with "newBook": the book as an object && "addBookToUser": the **UserObject** => from now on we will refer to value (object) of "newBook" as **BookObkect**
+
+### /books/:id
+
+- Get Book
+  **get**
+  _Backend Response_ => **BookObject**
+
+- Update one/many Fields In A Book
+  **put**
+  _Frontend Request_ => Any **BookObject** field that is not a _Mongoose ID_
+  _Notes_ => If the field is made up =>
+
+  ```
+  "This field does not exsist": "book data"
+  ```
+
+  Will **not** return an error, just the **BookObject** not changed in any way. If you try and change a field that is a _Mongoose ID_, you will recieve an error.
+  _Backend Response_ => **BookObject**
+  _Notes_ => The fields changed will be shown straight away in the response
+
+- Delete Book
+  **del**
+  _Backend Response_ => **BookObject**
+
+### /books/user
+
+- Add Interested User To Book
+  **post**
+  _Frontend Request_ =>
+  ```
+  {
+    "userId": "mongoose id",
+    "bookId": "mongoose id"
+  }
+  ```
+  _Notes_ => This will happen when the user swipes that they like a book.
+  _Backend Response_ => **BookObject**
+  _Notes_ => Currently in the process of making a middleware that can stop the duplication of interested user, however right now you can add the same user to a book multiple times.
```

Comment:
did you set yourself a ticket to update the description after we provide the middleware?

Reply:
or do they check it in the fronend?

Reply:
both, i just set up a middleware route, i will add to librix jira, i think after every update we need to update the notes with it

Reply:
allright